src/filesystem.tsx

```diff
@@ -88,3 +90,11 @@ function sortedFileEntries(entries: FileEntry[]): FileEntry[] {
       children: entry.children ? sortedFileEntries(entry.children) : entry.children,
     }));
 }
+
+function parsePdfIngestionResponse(response: PdfIngestionResponse): ReferenceItem[] {
+  return response.references.map((reference) => ({
+    id: reference.filename_md5,
+    title: reference.title ?? reference.source_filename.replace('.pdf', ''),
+    authors: reference.authors.map((author) => ({ fullName: author.full_name, surname: author.surname })),
+  }));
+}
```

Comment:
We should move the PDF Ingestion related code to outside of the `filesystem.tsx`

Reply:
I definitely agree, this will be done as part of #75 rather than in this PR